python/lsst/pipe/tasks/diffractionSpikeMask.py

```diff
@@ -0,0 +1,436 @@
+
+
+class DiffractionSpikeMaskConfig(Config):
+    """Config for BrightStarMaskTask.
+    """
+    refObjLoader = ConfigField(dtype=LoadReferenceObjectsConfig,
+                               doc="Configuration of reference object loader")
+    applyColorTerms = Field(
+        dtype=bool,
+        default=False,
+        doc=("Apply photometric color terms to reference stars?\n"
+             "`True`: attempt to apply color terms; fail if color term data is "
+             "not available for the specified reference catalog and filter.\n"
+             "`False`: do not apply color terms."),
+        optional=True,
+    )
+    colorterms = ConfigField(
+        dtype=ColortermLibrary,
+        doc="Library of photometric reference catalog name: color term dict"
+        " (see also applyColorTerms).",
+    )
+    photoCatName = Field(
+        dtype=str,
+        optional=True,
+        doc=("Name of photometric reference catalog; used to select a color"
+             " term dict in colorterms. See also applyColorTerms."),
+    )
+    raKey = Field(
+        dtype=str,
+        default="coord_ra",
+        doc="RA column name in the reference catalog.",
+    )
+    decKey = Field(
+        dtype=str,
+        default="coord_dec",
+        doc="Declination column name in the reference catalog.",
+    )
+    angleMargin = Field(
+        dtype=float,
+        default=60.,
+        doc="Margin outside the exposure bounding box to include bright "
+        "sources. In arcseconds.",
+    )
+    magnitudeThreshold = Field(
+        dtype=float,
+        default=15,
+        doc="Threshold magnitude for treating a star from the reference catalog"
+        " as bright.",
+    )
+    diffractAngle = Field(
+        dtype=float,
+        default=45,
+        doc="Angle in degrees of the location of diffraction spikes with "
+        "respect to camera at 0 rotation angle.",
+    )
+    spikeAspectRatio = Field(
+        dtype=float,
+        default=10,
+        doc="Ratio of the length of a diffraction spike to it's width in the"
+        " core of the star.",
+    )
+    magSlope = Field(
+        dtype=float,
+        default=-0.12,
+        doc="Slope of the fit for the log(spike length) as a function of"
+        " magnitude.",
+    )
+    magOffset = Field(
+        dtype=float,
+        default=3.8,
+        doc="Intercept of the fit for the log(spike length) as a function of"
+        " magnitude.",
+    )
+    fallbackMagnitude = Field(
+        dtype=float,
+        default=12.,
+        doc="Default magnitude to use for sources in the reference catalog"
+        " with missing magnitudes that land in regions of saturated pixels.",
+    )
+    anyFilterMapsToThis = Field(
+        dtype=str,
+        default="phot_g_mean",
+        doc="Fallback flux field in the reference catalog to use for sources"
+        " that don't have measurements in the science image's band."
+        "The default uses Gaia g-band.",
```

Comment:
I don't think that's what this config means.  Also don't put the default in the doc string!  Also also it's Gaia G *not* gaia g.